addOns/exim/src/main/java/org/zaproxy/addon/exim/AbstractPopupMenuSaveMessage.java

```diff
@@ -0,0 +1,259 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2021 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.exim;
+
+import java.io.File;
+import javax.swing.JFileChooser;
+import javax.swing.JMenu;
+import javax.swing.filechooser.FileFilter;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.model.Model;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.view.View;
+import org.zaproxy.zap.view.popup.PopupMenuHttpMessageContainer;
+import org.zaproxy.zap.view.popup.PopupMenuItemHttpMessageContainer;
+import org.zaproxy.zap.view.widgets.WritableFileChooser;
+
+public abstract class AbstractPopupMenuSaveMessage extends PopupMenuHttpMessageContainer {
+
+    private static final long serialVersionUID = 8080417865825721164L;
+
+    public static enum MessageComponent {
+        REQUEST,
+        REQUEST_HEADER,
+        REQUEST_BODY,
+        RESPONSE,
+        RESPONSE_HEADER,
+        RESPONSE_BODY
+    }
+
+    public AbstractPopupMenuSaveMessage(
```

Comment:
`protected`

Reply:
"Illegal modifier for the class AbstractPopupMenuSaveMessage; only public, abstract & final are permitted"

Reply:
Did you replace in the constructor?